custom/ewsghana/reports/email_reports.py

```diff
@@ -0,0 +1,222 @@
+            DataTablesColumn(_('Total # Facilities Registered with this Product')),
+            DataTablesColumn(_('Total Stock')),
+            DataTablesColumn(_('% Facilities with Consumption Data')),
+            DataTablesColumn(_('Monthly Consumption')),
+            DataTablesColumn(_('Months of Stock')),
+            DataTablesColumn(_('Stock Status'))
+        ])
+
+    @property
+    def rows(self):
+        def percent(x, y):
+            return "%d%% (%d)" % (x * 100 / (y or 1), x)
+
+        def stock_status(status):
+            stock_levels = CommtrackConfig.for_domain(self.config['domain']).stock_levels_config
+            if status == 0.0:
+                return 'Stockout'
+            elif status < stock_levels.understock_threshold:
+                return 'Low'
+            elif status < stock_levels.overstock_threshold:
+                return 'Adequate'
+            else:
+                return 'Overstock'
+
+        locations = self.get_locations(self.config['location_id'], self.config['domain'])
+
+        products = self.unique_products(locations)
+        row_data = {
+            product.name: dict(
+                fac_with_stockout=0,
+                total_fac=0,
+                total_stock=0,
+                fac_with_consumption=0,
+                monthly_consumption=0
+            ) for product in products
+        }
+
+        for location in locations:
+            stock_states = StockState.objects.filter(
+                case_id=location.supply_point_id,
+                section_id=STOCK_SECTION_TYPE,
+                sql_product__in=products
+            ).order_by('-last_modified_date')
+
+            for state in stock_states:
+                p_name = state.sql_product.name
+                row_data[p_name]['total_fac'] += 1
+                if state.stock_on_hand:
+                    row_data[p_name]['total_stock'] += state.stock_on_hand
+                else:
+                    row_data[p_name]['fac_with_stockout'] += 1
+                if state.get_monthly_consumption():
+                    row_data[p_name]['fac_with_consumption'] += 1
+                    row_data[p_name]['monthly_consumption'] += state.get_monthly_consumption()
+
+        rows = []
+        for k, v in row_data.iteritems():
+            months_of_stock = float(v['total_stock']) / float(v['monthly_consumption'] or 1)
+            rows.append([k, percent(v['fac_with_stockout'], v['total_fac']),
+                        v['total_fac'], v['total_stock'], percent(v['fac_with_consumption'], v['total_fac']),
+                        v['monthly_consumption'], "%.1f" % months_of_stock, stock_status(months_of_stock)])
+        return rows
+
+
+class StockSummaryReportData(EmailReportData):
+    title = 'Weekly Stock Summary Report - '
+    slug = 'stock_summary'
+
+    def get_locations(self, loc_id, domain):
+        self.title += SQLLocation.objects.get(location_id=loc_id).name
```

Comment:
does `self.title` need to be writeable? this seems like a funny place to do this work, and it would be cleaner if you made it a property. (applies to other reports)